src/zone/scheduler.h

```diff
@@ -111,7 +118,13 @@ namespace zone {
         NAPA_DEBUG("Scheduler", "Shutting down: Start draining unscheduled tasks...");
 
         // Wait for all tasks to be scheduled.
-        while (_beingScheduled > 0 || !_nonScheduledTasks.empty()) {
+        while (_beingScheduled > 0
```

Comment:
Note this usage of _beijingScheduled !  which means in scheduleAll need to ++ it before send to synchronizer.

Reply:
_beingScheduled is designed to indicate the number os tasks being scheduled nut not yet dispatched to worker. Actually, ScheduleAll has assigned the task to all workers. I think that's why current ScheduleOnWorker and ScheduleAll don't change _beingScheduled.

Reply:
updated, as well as the original comment of _beingScheduled which is not accurate.